### PR TITLE
add CorsInterceptorTest

### PR DESCRIPTION
**Pull request by jschneid-nmdp**
This pull request is related to a recent clinFHIR enhancement to enable use of an OAuth access token with clinFHIR, as discussed on chat.fhir.org here:

https://chat.fhir.org/#narrow/stream/179176-clinFHIR/topic/OAuth.20access.20token

When using an OAuth access token, clinFHIR adds an "Authorization: Bearer <access_token>" header to its FHIR API requests.  Additionally, it sends a CORS preflight HTTP OPTIONS request.

When testing this using a FHIR server based on hapi-fhir v5.1.0, David Hay and I discovered the server was returning HTTP status 403 (Forbidden) in its response to the CORS preflight OPTIONS request.

To prevent the server from rejecting the CORS preflight OPTIONS request, instead of using the default CorsInterceptor configuration, it was necessary to customize the CorsConfiguration.

In short, this CorsInterceptor doesn't work:
```
       CorsInterceptor corsInterceptor = new CorsInterceptor();
```

But this does:
```
        CorsConfiguration corsConfiguration = new CorsConfiguration();
        corsConfiguration.applyPermitDefaultValues();
        corsConfiguration.setAllowedMethods(Arrays.asList(new String[] { "*" }));
        CorsInterceptor corsInterceptor = new CorsInterceptor(corsConfiguration);
```

This pull request adds a CorsInterceptorTest which may be useful for exploring the CorsConfiguration topic.  (I wasn't sure what modification to the default CorsConfiguration might be most appropriate, but it appears the "Authorization" header is not among the allowedHeaders in the default CorsConfiguration.)

### CODE REVIEW

**Comment by jschneid-nmdp** [comment]
The test's System.error output looks like this:
```
Default CorsConfiguration
allowCredentials = null
allowedHeaders = [Origin, X-FHIR-Starter, Accept, Access-Control-Request-Method, Cache-Control, X-Requested-With, Access-Control-Request-Headers, Prefer, Content-Type]
allowedMethods = [DELETE, POST, GET, OPTIONS, PUT, PATCH]
allowedOrigins = [*]
exposedHeaders = [Content-Location, Location]
maxAge = null
Custom CorsConfiguration
allowCredentials = null
allowedHeaders = [*]
allowedMethods = [*]
allowedOrigins = [*]
exposedHeaders = null
maxAge = 1800
```

**Comment by jschneid-nmdp** [comment]
It would be best to also add relevant exposed headers to the custom CorsConfiguration, e.g.:
```
		corsConfiguration.setExposedHeaders(Arrays.asList(new String[] {
			"Content-Location",
			"Date",
			"ETag",
			"Location",
			"X-Request-Id",
			"X-Correlation-Id"
		}));
```

File: hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/CorsInterceptorTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			"Content-Location",
+			"Date",
+			"ETag",
+			"Location",
+			"X-Request-Id",
+			"X-Correlation-Id"
+		}));
+		CorsInterceptor corsInterceptor = new CorsInterceptor(corsConfiguration);
+
+System.err.println("Custom CorsConfiguration");
```
Comment (jamesagnew) [inline hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/CorsInterceptorTest.java:30]: Output should be done with a logger, e.g. `ourLog.info("Custom CorsConfiguration");`. See other tests in this project for examples.

**Review by jamesagnew — CHANGES_REQUESTED** [review]
Looks great, thanks for the contribution! If you wouldn't mind to replace the logging, I'm happy to merge.

File: hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/CorsInterceptorTest.java (hunk 1 above)
Comment (jschneid-nmdp) [reply hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/CorsInterceptorTest.java]: Ok, done!

I revised the code to use SLF4J Logger.info() instead of System.err.println(), and also modified the code to run the test assertions before executing the potentially brittle log output code.

Additionally, in pom.xml I added a test-scoped logback-classic dependency, because SLF4J was defaulting to a NOP logger implementation without it (see also http://www.slf4j.org/codes.html#StaticLoggerBinder).

**Review by jamesagnew — APPROVED** [review]

File: hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/CorsInterceptorTest.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-server/src/test/java/ca/uhn/fhir/rest/server/interceptor/CorsInterceptorTest.java]: Thanks! Merging now
